enhancements/cluster-api/installing-openshift-natively-via-cluster-api.md

```diff
@@ -0,0 +1,598 @@
+
+Cluster API is a Kubernetes sub-project that provides a declarative API for managing the lifecycle of Kubernetes clusters and the infrastructure behind them.
+It provides a multi-cluster management solution that will enable OpenShift to unify the single cluster and multi-cluster installation experiences.
+
+[^3]: Leading to various UPI clusters and other installer variations.
+
+### User Stories
+
+As an OpenShift administrator, I want to be able to provision my first OpenShift cluster using the same tooling that I use to provision additional clusters, so that I can have a consistent experience.
+
+As an OpenShift administrator, I want to be able to centrally manage the lifecycle and infrastructure of my OpenShift clusters, so that I do not have to delegate infrastructure management to other teams.
+
+As an OpenShift developer, I want to leverage Cluster API to provision OpenShift cluster infrastructure, so that I can focus on the OpenShift specific components and leverage the community around Cluster API for the generic cluster Infrastructure.
+
+As an OpenShift developer, I want to simplify the maintenance of the installer by removing the embedded terraform and platform specific logic, so that the installer is easier to maintain and extend.
+
+As a Hive developer, I want to delegate as much of the cluster lifecycle to Cluster API as possible, so that Hive can focus on the OpenShift specific features.
+
+### Goals
+
+#### Phase 1
+
+1. Leverage an existing Cluster API control plane to provision infrastructure for OpenShift clusters.
+1. Delegate the provisioning and management of the cluster infrastructure bootstrap provisioning to Cluster API.
+1. Delegate the orchestration of the OpenShift cluster bootstrap node process to Cluster API.
+1. Supplement Cluster API infrastructure provisioning gaps with small, and focused supplemental infrastructure providers.
+1. Convert the installer to generate Cluster API infrastructure templates instead of terraform resources.
+1. Rely on the installer to generate configuration up to and including the `manifests` phase.
+1. Leverage the desired payload installer to generate bootstrap ignition configs and kubeconfigs.
+
+#### Phase 2
+
+1. Provide an end to end OpenShift installation experience using Cluster API.
+  1. This will allow a BYO infrastructure experience for OpenShift via Cluster API.
+1. Integrate Hive directly with Cluster API to remove the need to execute the installer during Cluster Deployments.
+
+### Non-Goals
+
+1. Provisioning of the Cluster API control plane.[^2]
+1. Generation in-tree of ignition data for the bootstrap node.
+
+## Proposal
+
+A Cluster API control plane and bootstrap provider will be created to handle the orchestration and configuration of the OpenShift cluster during the bootstrap process.
```

Comment:
What are these providers? Are they controllers?

Reply:
Yeah, the idea is building small controllers that handle the different parts of the bootstrap process and report back via status objects